site/app/controllers/admin/UsersController.php

```diff
@@ -1108,11 +1063,17 @@ public function uploadUserList($list_type = "classlist") {
                 $user->setLegalFirstName($row[1]);
                 $user->setLegalLastName($row[2]);
                 $user->setEmail($row[3]);
-                $set_user_registration_or_group_function($user);
-                if (isset($row[$pref_firstname_idx]) && !empty($row[$pref_firstname_idx])) {
+                // Registration section has to exist, or a DB exception gets thrown on INSERT or UPDATE.
+                // ON CONFLICT clause in DB query prevents thrown exceptions when registration section already exists.
+                if (!empty($row[$registration_section_idx])) {
+                    $this->core->getQueries()->insertNewRegistrationSection($row[$registration_section_idx]);
+                    $user->setRegistrationSection($row[$registration_section_idx]);
+                }
+                $user->setGroup($list_type === 'classlist' ? 4 : $row[4]);
+                if (!empty($row[$pref_firstname_idx])) {
                     $user->setPreferredFirstName($row[$pref_firstname_idx]);
                 }
-                if (isset($row[$pref_lastname_idx]) && !empty($row[$pref_lastname_idx])) {
+                if (!empty($row[$pref_lastname_idx])) {
```

Comment:
Explanation: In lines 975 + 981 and 1073 + 1076 (they are related pieces of logic), using both `isset` and `!empty` for preferred name entries seemed redundant to me since `!empty` is more exclusive than `isset`. (I noticed since I also had to use `!empty` to check the registration section entry, so I made this change along the way.)